crates/block-producer/src/rpc_client.rs

```diff
@@ -680,7 +679,7 @@ impl RPCClient {
             )?;
 
             if cells.last_cursor.is_empty() {
-                return Err(anyhow!("no enough custodian cells"));
+                return Ok(collected);
```

Comment:
Why return Ok when we can't get enough custodian cells?

Reply:
The purpose of this function is changed, the result will be also be used to filter withdrawal requests.
Even if the total balance isn't enough, we can still process some of them. For example, A withdraw 200, B withdraw 400.
But we only have 300 finalized available at the time, we can still process A request and skip B.

Reply:
Since we gather finalized custodians info from db, restore this function.